From: Andrew Morton <akpm@linux-foundation.org>
Subject: kprobeslib-kretprobe-scalability-improvement-fix
Date: Wed Nov  2 03:25:47 PM PDT 2022

fix build with -Werror=designated-init

lib/test_objpool.c:509:10: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
  509 |         {ot_init_sync_m0, ot_fini_sync_m0},
      |          ^~~~~~~~~~~~~~~
lib/test_objpool.c:509:10: note: (near initialization for 'g_ot_sync_ops[0]')
lib/test_objpool.c:509:10: error: invalid initializer
lib/test_objpool.c:509:10: note: (near initialization for 'g_ot_sync_ops[0].<anonymous>')

Cc: Anil S Keshavamurthy <anil.s.keshavamurthy@intel.com>
Cc: Dan Williams <dan.j.williams@intel.com>
Cc: David S. Miller <davem@davemloft.net>
Cc: Eric Biggers <ebiggers@google.com>
Cc: Josh Poimboeuf <jpoimboe@kernel.org>
Cc: Masami Hiramatsu (Google) <mhiramat@kernel.org>
Cc: "Naveen N. Rao" <naveen.n.rao@linux.ibm.com>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Sander Vanheule <sander@svanheule.net>
Cc: Steven Rostedt (Google) <rostedt@goodmis.org>
Cc: wuqiang <wuqiang.matt@bytedance.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/test_objpool.c |   16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

--- a/lib/test_objpool.c~kprobeslib-kretprobe-scalability-improvement-fix
+++ a/lib/test_objpool.c
@@ -506,10 +506,10 @@ struct {
 	struct ot_context * (*init)(void);
 	void (*fini)(struct ot_context *sop);
 } g_ot_sync_ops[4] = {
-	{ot_init_sync_m0, ot_fini_sync_m0},
-	{ot_init_sync_m1, ot_fini_sync_m1},
-	{ot_init_sync_m2, ot_fini_sync_m2},
-	{ot_init_sync_m3, ot_fini_sync_m3},
+	{.init = ot_init_sync_m0, .fini = ot_fini_sync_m0},
+	{.init = ot_init_sync_m1, .fini = ot_fini_sync_m1},
+	{.init = ot_init_sync_m2, .fini = ot_fini_sync_m2},
+	{.init = ot_init_sync_m3, .fini = ot_fini_sync_m3},
 };
 
 /*
@@ -810,10 +810,10 @@ struct {
 	struct ot_context * (*init)(void);
 	void (*fini)(struct ot_context *sop);
 } g_ot_async_ops[4] = {
-	{ot_init_async_m0, ot_fini_async},
-	{ot_init_async_m1, ot_fini_async},
-	{ot_init_async_m2, ot_fini_async},
-	{ot_init_async_m3, ot_fini_async},
+	{.init = ot_init_async_m0, .fini = ot_fini_async},
+	{.init = ot_init_async_m1, .fini = ot_fini_async},
+	{.init = ot_init_async_m2, .fini = ot_fini_async},
+	{.init = ot_init_async_m3, .fini = ot_fini_async},
 };
 
 static void ot_nod_recycle(struct ot_node *on, struct objpool_head *pool,
_
